### ISSUE TITLE
.NET7:  Nullable `Vector2?` field crashes the Editor

### ISSUE THREAD

**Issue by brmassa** [FlaxEngine/FlaxEngine#935]
**Issue Description:**
Actors/Prefabs that have a nullable Vector2 field `Vector2? variable;` crashes the Editor when trying to initialize it.

**Steps to reproduce:**

1* create a Script with the following code

```cs
using FlaxEngine;

[ActorContextMenu("MyClass")]
class MyClass : Actor
{
  public Vector2? myVariable;
}
```

2* create the actor (it will already throw exceptions, but the editor will stand)
3* create a prefab from this actor
4* open the prefab
5* crash

**Flax version:**
dotnet7

**Comment by mafiesto4** [comment]
```
GenericArguments[1], 'System.Nullable`1[FlaxEngine.Vector2]', on 'FlaxEngine.NativeInterop+MarshalHelper`1+ValueTypeField`1[T,TField]' violates the constraint of type 'TField'.
Stack strace:
   at System.RuntimeType.ValidateGenericArguments(MemberInfo definition, RuntimeType[] genericArguments, Exception e)
   at System.RuntimeType.MakeGenericType(Type[] instantiation)
   at FlaxEngine.NativeInterop.MarshalHelper`1..cctor() in F:\FlaxEngineDotnet\Source\Engine\Engine\NativeInterop.cs:line 1384
```

**Comment by mafiesto4** [comment]
Fixed in 609217a3bbada219e4029b3183d22b06e371f4e0
